novawallet/Modules/WalletsList/Common/WalletView.swift

```diff
@@ -0,0 +1,114 @@
+import UIKit
+
+final class WalletView: GenericTitleValueView<WalletIconView, GenericPairValueView<IconDetailsView, GenericPairValueView<UILabel, IconDetailsView>>> {
+    private var viewModel: ViewModel?
+
+    var iconImageView: UIImageView { titleView.iconViewImageView }
+    var networkImageView: UIImageView { titleView.networkIconImageView }
+
+    var titleLabel: UILabel { valueView.fView.detailsLabel }
+    var indicatorImageView: UIImageView { valueView.fView.imageView }
+    var subtitleLabel: UILabel { valueView.sView.fView }
+    var subtitleDetailsImage: UIImageView { valueView.sView.sView.imageView }
+    var subtitleDetailsLabel: UILabel { valueView.sView.sView.detailsLabel }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setupStyle()
+    }
+
+    func setupStyle() {
+        titleLabel.apply(style: .regularSubhedlinePrimary)
+        titleLabel.numberOfLines = 1
+        titleLabel.lineBreakMode = .byTruncatingTail
+        spacing = 12
+        alignment = .left
+        indicatorImageView.snp.makeConstraints {
+            $0.height.width.equalTo(8)
+        }
+        indicatorImageView.layer.cornerRadius = 4
+        indicatorImageView.backgroundColor = R.color.colorIconAccent()!
+        indicatorImageView.isHidden = true
+        subtitleLabel.apply(style: .footnoteSecondary)
+        valueView.sView.sView.iconWidth = 16
+        subtitleDetailsLabel.apply(style: .footnotePrimary)
+        valueView.fView.mode = .detailsIcon
+        valueView.sView.makeHorizontal()
+        valueView.sView.spacing = 4
+        subtitleLabel.setContentCompressionResistancePriority(.high, for: .horizontal)
+    }
+}
+
+extension WalletView {
+    struct ViewModel: Hashable, Equatable {
+        let icon: IdentifiableImageViewModelProtocol?
+        let networkIcon: IdentifiableImageViewModelProtocol?
+        let name: String
+        let subtitle: String
+        let subtitleDetailsIcon: IdentifiableImageViewModelProtocol?
```

Comment:
It is hard to understand for which wallet types which fields are used. I would make enum instead as we discussed: regular and proxy. Each case will contain only fields it is used